src/Context/UserInfo.tsx

```diff
@@ -0,0 +1,115 @@
+import { createContext, useContext, useReducer } from "react";
+import { ICar, IUserContext } from "../interfaces/User";
+
+const initialState = {
+  name: "Gerardo",
+  lastName: "Estrada",
+  isLoggedIn: false,
+  cars: [
+    {
+      brand: "Toyota",
+      color: "Blue",
+    },
+    {
+      brand: "Nissan",
+      color: "Red",
+    },
+  ],
+};
+
+export const UserContext = createContext<IUserContext>(initialState);
+
+export const useUserContext = () => {
+  const context = useContext(UserContext);
+  if (!context) {
+    throw new Error("You can't use the themecontext outside of the provider");
+  }
+  return context;
+};
+
+enum ActionKind {
+  Add = "ADD_TO_CAR",
+  Update = "UPDATE_THE_CAR",
+  Remove = "REMOVE_TO_CART",
+}
+
+type ActionCar = {
+  type: ActionKind;
+  payload: { newData?: ICar; oldData?: ICar };
+};
+
+const reducer = (state: IUserContext, action: ActionCar) => {
+  const { type, payload } = action;
+
+  switch (type) {
+    case ActionKind.Add:
+      if (payload.newData) {
+        state.cars.push(payload.newData);
+      }
+      return state;
+    case ActionKind.Update:
+      if (payload.newData && payload.oldData) {
+        let count = 0;
+        let index = 0;
+        state.cars.forEach((car) => {
+          if (car.brand === payload.oldData?.brand) {
+            index = count;
+          }
+          count++;
+        });
+        state.cars[index] = payload.newData;
+      }
+      return state;
+    case ActionKind.Remove:
+      if (payload.oldData) {
+        state.cars = state.cars.filter((car) => {
+          return car.brand !== payload.oldData?.brand;
+        });
+      }
+      return state;
```

Comment:
Same important suggestion, always avoid mutating state. Also, you don't need to pass the entire object of the car you want to remove, instead you can only pass the ID
```suggestion
      return {
          ...state,
          cars: state.cars.filter((car) => car.id !== payload) // This would work if the payload is the id only
      }
```